codegenerator/cli/templates/static/codegen/src/bindings/BigInt.res

```diff
@@ -43,14 +44,17 @@ module Bitwise = {
 
 let zero = fromInt(0)
 
-let schema =
-  S.string
-  ->S.setName("BigInt")
-  ->S.transform(s => {
-    parser: (. string) =>
-      switch string->fromString {
-      | Some(bigInt) => bigInt
-      | None => s.fail(. "The string is not valid BigInt")
-      },
-    serializer: (. bigint) => bigint->toString,
-  })
+//Note this schema needs to be able to parse unknowns since it's used in the rollback
+//function. Postgres encodes large numerics as numbers with exponents and we encode as strings
+//so we need to be able to parse both.
+let schema: S.t<bigint> = S.custom("BigInt", s => {
+  parser: unknown => {
+    switch fromUnknownUnsafe(unknown) {
+    | exception _ => s.fail("Not a valid BigInt")
+    | b => b
+    }
+  },
+  serializer: b => {
+    b->toString
+  },
+})
```

Comment:
I'm using the built in `row_to_json` function. This serialises large numeric types like this:
```json
{ "balance": 9e+24, "id": "0x177Df24aDDC9a216f927D2A894aB0b6EeC59Eb09"}
```

So I needed a way for the parser to accept both representations without throwing a validation error because it's not a string.

Reply:
Makes sense. I'm going to add it as a built-in type in the next rescript-schema release

Reply:
Shouldn't it be a float in this case?

Reply:
> Shouldn't it be a float in this case?

I'm not sure I understand? Why would it be a float?

Reply:
> Or if the `row_to_json` converts bigint string values to numbers, then we should find another solution because it'll cause data loss

We use pg numeric with variable precision for bigints. Not strings. When pg serializes it outputs the exponent notation. This value is valid for constructing a bigint with  BigInt(...) constructor. 

All we're doing is taking schema of bigint and now able to parse it when it's serialized with exponent notation.

Reply:
If BigInt is more than the safe number, it'll lose data during the numeric -> float conversion

BigInt(Number(9000000000000000385875969n))
9000000000000000385875968n